sync/src/block/downloader/header.rs

```diff
@@ -100,12 +102,17 @@ impl HeaderDownloader {
         self.request_time.map_or(false, |time| (Instant::now() - time).as_secs() > MAX_WAIT)
     }
 
-    /// Find header from download cache, and then from blockchain
+    /// Find header from queued headers, downloaded cache and then from blockchain
     /// Panics if header dosn't exist
     fn pivot_header(&self) -> Header {
-        match self.downloaded.get(&self.pivot.hash) {
+        match self.queued.get(&self.pivot.hash) {
             Some(header) => header.clone(),
-            None => self.client.block_header(&BlockId::Hash(self.pivot.hash)).unwrap(),
+            None => self
+                .downloaded
+                .get(&self.pivot.hash)
+                .map(Clone::clone)
+                .or_else(|| self.client.block_header(&BlockId::Hash(self.pivot.hash)))
+                .unwrap(),
```

Comment:
Why did you use function chaining here?
Using `match` seems to be more readable for me.

Reply:
I changed it to `match` style.